### PR TITLE
Add GameStats cog and add Apex Stats.

### PR DESCRIPTION
**Pull request by trilleplay**
This PR requires a API Key from: https://apex.tracker.gg/site-api inorder to be utilized.

### CODE REVIEW

File: GearBot/Cogs/GameStats.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+class GameStats(BaseCog):
+
+    def __init__(self, bot):
+        super().__init__(bot, {
+            "min": 0,
+            "max": 6,
+            "required": 0,
+            "commands": {}
+        })
+        self.session = aiohttp.ClientSession(loop=bot.loop)
```
Comment (AEnterprise) [inline GearBot/Cogs/GameStats.py]: this isn't needed, there is ``bot.aiosession``, that serves a client session for the entire bot

File: GearBot/Cogs/GameStats.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    @commands.command()
+    async def apexstats(self, ctx, platform, *, username):
+        """about_apexstats"""
+        if platform is None:
```
Comment (AEnterprise) [inline GearBot/Cogs/GameStats.py]: can't be none anymore now it's a required param

File: GearBot/Cogs/GameStats.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        super().__init__(bot, {
+            "min": 0,
+            "max": 6,
+            "required": 0,
+            "commands": {}
+        })
+        self.session = aiohttp.ClientSession(loop=bot.loop)
+
+    @commands.command()
+    async def apexstats(self, ctx, platform, *, username):
```
Comment (AEnterprise) [inline GearBot/Cogs/GameStats.py]: can you make a converter for platform? might be useful in other places, also please make it case insensitive and means you don't have to worry about if it's valid here or not, the converter can then raise the error and use the same parse errors the rest of the bot use

File: GearBot/Cogs/GameStats.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        elif platform == "xbox":
+            apexplatform = "1"
+        else:
+            await ctx.send(Translator.translate('apexstats_invalid_platform', ctx))
+            return
+        url = "https://public-api.tracker.gg/apex/v1/standard/profile/" + apexplatform + "/" + (username)
+        if not Configuration.get_master_var("APEX_KEY", "0") is "0":
+            headers = {"TRN-Api-Key": Configuration.get_master_var("APEX_KEY")}
+        else:
+            await ctx.send("There is no API key provided by the Administrator of this GearBot stats.")
```
Comment (AEnterprise) [inline GearBot/Cogs/GameStats.py]: hardcoded string, also wouldn't it be better to just kill the cog if the key is missing? like make the cog unload itself during loading if it's missing

File: GearBot/Cogs/GameStats.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+            return
+        async with self.session.get(url, headers=headers) as resp:
+            if resp.status == 404:
+                await ctx.send(Translator.translate('apexstats_user_not_found', ctx))
```
Comment (AEnterprise) [inline GearBot/Cogs/GameStats.py]: same as above, a no emoji would make it more consistent with the rest of the bot

File: GearBot/Cogs/GameStats.py [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+                return
+            elif not resp.status == 200:
+                await ctx.send(Translator.translate('apexstats_api_error', ctx))
```
Comment (AEnterprise) [inline GearBot/Cogs/GameStats.py]: same here

File: GearBot/Cogs/GameStats.py (hunk 3 above)
Comment (trilleplay) [reply GearBot/Cogs/GameStats.py]: Unresolved cause i forgot about the case insensitive part.

Comment (trilleplay) [reply GearBot/Cogs/GameStats.py]: should be fixed now.

File: lang/en_US.json [hunk 7]
```diff
@@ -624,6 +624,19 @@
...
+  "apexstats_api_error": "I was unable to retrive stats from that player. It's possible that the API provider might be having issues. If you keep getting this error please join the Support Server and let us know.",
+  "apexstats_key_Level": "Level",
+  "apexstats_key_Kills": "Kills",
+  "apexstats_key_KillsPerMatch": "Kills Per Match",
+  "apexstats_key_DamagePerMatch": "Damage Per Match",
+  "apexstats_key_KillsAsKillLeader": "Kills As Kill Leader",
+  "apexstats_key_Damage": "Damage",
+  "apexstats_key_MatchesPlayed": "Matches Played",
+  "apexstats_key_CarePackageKills": "Care Package Kills",
+  "apexstats_username": "Username",
```
Comment (Tarek1337) [inline lang/en_US.json]: I've been testing this a bit on my own bot, and there is a lot of strings missing, i.e shotgun kills, smg kills and so on.

Comment (trilleplay) [reply lang/en_US.json]: Tarek could you DM me the Key names, these was all keys i was able to find.

**Comment by Tarek1337** [comment]
Will do.

Den ons 3 apr. 2019 09:12trilleplay <notifications@github.com> skrev:

File: lang/en_US.json (hunk 7 above)
Comment (Tarek1337) [reply lang/en_US.json]: Sure thing.

Comment (trilleplay) [reply lang/en_US.json]: Added the keys you provided in https://github.com/AEnterprise/GearBot/pull/187/commits/2f62abdc0c323b7689a94de5ca41bf079bcf4bec

**Pull request merged**